pyOCD/flash/flash_builder.py

```diff
@@ -422,3 +504,106 @@ def _page_erase_program(self, progress_cb = _stub_progress):
+                data = self.flash.target.readBlockMemoryUnaligned8(page.addr, len(page.data))
+                page.same = _same(page.data, data)
+                progress += page.getVerifyWeight()
+                count += 1
+                if page.same:
+                    same_count += 1
+
+                # Update progress
+                progress_cb(float(progress) / float(self.page_erase_weight))
+        logging.debug("scanned %d pages for equality (%d same)", count, same_count)
+        return progress
+
+    def _next_nonsame_page(self, i):
+        if i >= len(self.page_list):
+            return None, i
+        page = self.page_list[i]
+        while page.same:
+            i += 1
+            if i >= len(self.page_list):
+                return None, i
+            page = self.page_list[i]
+        return page, i+1
+
+    def _page_erase_program_double_buffer(self, progress_cb = _stub_progress):
+        """
+        Program by performing sector erases.
+        """
+        actual_page_erase_count = 0
+        actual_page_erase_weight = 0
+        progress = 0
+
+        progress_cb(0.0)
+
+        # Fill in same flag for all pages. This is done up front so we're not trying
+        # to read from flash while simultaneously programming it.
+        progress = self._scan_pages_for_same(progress_cb)
+
+        # Set up page and buffer info.
+        error_count = 0
+        current_buf = 0
+        next_buf = 1
+        page, i = self._next_nonsame_page(0)
+
+        # Make sure there are actually pages to program differently from current flash contents.
+        if page is not None:
+            # Load first page buffer
+            self.flash.loadPageBuffer(current_buf, page.addr, page.data)
+
+            while page is not None:
+                assert page.same is not None
+
+                # Kick off this page program.
+                current_addr = page.addr
+                current_weight = page.getEraseProgramWeight()
+                self.flash.erasePage(current_addr)
+                self.flash.startProgramPageWithBuffer(current_buf, current_addr) #, erase_page=True)
+                actual_page_erase_count += 1
+                actual_page_erase_weight += page.getEraseProgramWeight()
+
+                # Get next page and load it.
+                page, i = self._next_nonsame_page(i)
+                if page is not None:
+                    self.flash.loadPageBuffer(next_buf, page.addr, page.data)
+
+                # Wait for the program to complete.
+                result = self.flash.waitForCompletion()
+
+                # check the return code
+                if result != 0:
+                    logging.error('programPage(0x%x) error: %i', current_addr, result)
+                    error_count += 1
+                    if error_count > self.max_errors:
+                        logging.error("Too many page programming errors, aborting program operation")
+                        break
```

Comment:
I only saw errors when I did something wrong while working on the code. :) I figured it wouldn't hurt to be a little forgiving. Though we should probably be returning a status code to the caller.